From 930ea38e178c9196e599fe4f55e207bc5973c4c4 Mon Sep 17 00:00:00 2001
From: Fabio Erculiani <lxnay@sabayon.org>
Date: Tue, 4 Jan 2011 22:54:22 +0100
Subject: [PATCH 1/5] Change default sleep time when CDROOT=1 and sdelay is not set.

Many people reported that sleeping for just 1 second, when booting
off a livecd is not enough to guarantee that the drive and disc
inside it, gets initialized in time.
---
 defaults/initrd.scripts |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/defaults/initrd.scripts b/defaults/initrd.scripts
index e0710c4..fe9fdd1 100755
--- a/defaults/initrd.scripts
+++ b/defaults/initrd.scripts
@@ -879,6 +879,13 @@ sdelay() {
 	if [ -n "${SDELAY}" ]
 	then
 		sleep ${SDELAY}
+	elif [ "${CDROOT}" = '1' ]
+	then
+		# many CD/DVD drives require extra sleep, especially when
+		# connected via USB. Many people reported that 1 second sleep
+		# is not enough on their notebooks, that's why sleep 3, when booting
+		# off a livecd is a better default.
+		sleep 3
 	else
 		sleep 1
 	fi
-- 
1.7.3.2

